nengo_loihi/hardware/tests/test_allocators.py

```diff
@@ -57,4 +61,180 @@ def test_one_to_one_allocator_big_block_error():
+    board = allocator(model)
+
+    assert board.n_chips == 1
+    assert board.n_cores_per_chip == [3]
+    assert board.n_synapses_per_core == [[1, 1, 0]]
+
+    chip = board.chips[0]
+    assert chip.board is board
+    assert chip.n_cores == 3
+
+    assert chip.cores[0].chip is chip
+    assert len(chip.cores[0].synapses) == 1
+    assert len(chip.cores[0].blocks) == 1
+    assert len(chip.cores[0].inputs) == 0
+
+    assert chip.cores[1].chip is chip
+    assert len(chip.cores[1].synapses) == 1
+    assert len(chip.cores[1].blocks) == 1
+    assert len(chip.cores[1].inputs) == 0
+
+    assert chip.cores[2].chip is chip
+    assert len(chip.cores[2].synapses) == 0
+    assert len(chip.cores[2].blocks) == 0
+    assert len(chip.cores[2].inputs) == 1
+
+
+def test_round_robin_allocator_under():
+    model = _basic_model()
+
+    board = RoundRobin(n_chips=2)(model)
+
+    assert board.n_chips == 2
+    assert board.n_cores_per_chip == [2, 1]
+    assert board.n_synapses_per_core == [[1, 0], [1]]
+
+    chip0 = board.chips[0]
+    assert chip0.board is board
+    assert chip0.n_cores == 2
+
+    assert chip0.cores[0].chip is chip0
+    assert len(chip0.cores[0].synapses) == 1
+    assert len(chip0.cores[0].blocks) == 1
+    assert len(chip0.cores[0].inputs) == 0
+
+    assert chip0.cores[1].chip is chip0
+    assert len(chip0.cores[1].synapses) == 0
+    assert len(chip0.cores[1].blocks) == 0
+    assert len(chip0.cores[1].inputs) == 1
+
+    chip1 = board.chips[1]
+    assert chip1.board is board
+    assert chip1.n_cores == 1
+
+    assert chip1.cores[0].chip is chip1
+    assert len(chip1.cores[0].synapses) == 1
+    assert len(chip1.cores[0].blocks) == 1
+    assert len(chip1.cores[0].inputs) == 0
+
+
+def test_round_robin_allocator_over():
+    model = _basic_model()
+
+    board = RoundRobin(n_chips=4)(model)
+
+    assert board.n_chips == 3
+    assert board.n_cores_per_chip == [1, 1, 1]
+    assert board.n_synapses_per_core == [[1], [1], [0]]
+
+    chip0 = board.chips[0]
+    assert chip0.board is board
+    assert chip0.n_cores == 1
+
+    assert chip0.cores[0].chip is chip0
+    assert len(chip0.cores[0].synapses) == 1
+    assert len(chip0.cores[0].blocks) == 1
+    assert len(chip0.cores[0].inputs) == 0
+
+    chip1 = board.chips[1]
+    assert chip1.board is board
+    assert chip1.n_cores == 1
+
+    assert chip1.cores[0].chip is chip1
+    assert len(chip1.cores[0].synapses) == 1
+    assert len(chip1.cores[0].blocks) == 1
+    assert len(chip1.cores[0].inputs) == 0
+
+    chip2 = board.chips[2]
+    assert chip2.board is board
+    assert chip2.n_cores == 1
+
+    assert chip2.cores[0].chip is chip2
+    assert len(chip2.cores[0].synapses) == 0
+    assert len(chip2.cores[0].blocks) == 0
+    assert len(chip2.cores[0].inputs) == 1
+
+
+@pytest.mark.skipif(pytest.config.getoption('--target') != 'loihi',
+                    reason="Hardware allocators are Loihi-only")
+def test_deterministic_network_allocation(Simulator, seed):
```

Comment:
Does this actually test that the network allocation is deterministic? It tests that the results are deterministic, which is good. But things could be being put on different chips/cores from one call of the allocator to the next and still have deterministic simulation. Maybe deterministic allocation isn't something we care about, as long as we have deterministic results. I just want to make sure it's clear what this test is actually testing.

Reply:
Added a comment to the test. Determinism of the allocation itself is addressed by some of the other tests.